app/src/main/java/org/jellyfin/androidtv/integration/LeanbackChannelWorker.kt

```diff
@@ -138,6 +148,69 @@ class LeanbackChannelWorker(
 		}.toTypedArray())
 	}
 
+	/**
+	 * Updates the "next up" row with current episodes
+	 */
+	private suspend fun updateNextUp() {
+		// Get channel
+		val channelUri = getChannelUri("next_up", Channel.Builder()
+			.setType(TvContractCompat.Channels.TYPE_PREVIEW)
+			.setDisplayName(context.getString(R.string.lbl_next_up))
+			.setAppLinkIntent(Intent(context, StartupActivity::class.java))
+			.build())
+
+		// Get user or return if no user is found (not authenticated)
+		val user = TvApp.getApplication().currentUser ?: return
+
+		val response = apiClient.getNextUpEpisodes(NextUpQuery().apply {
+			userId = user?.id
+			imageTypeLimit = 1
+			limit = 15
+		})
+
+		// Add new items
+		response?.items?.map { item ->
+			val imageUri = if (preferParentThumb && item.parentThumbItemId != null) {
+				Uri.parse(apiClient.GetImageUrl(item.seriesId, ImageOptions().apply {
+					format = ImageFormat.Png
+					height = 288
+					width = 512
+					imageType = ImageType.Thumb
+				}))
+			} else {
+				Uri.parse(apiClient.GetImageUrl(item, ImageOptions().apply {
+					format = ImageFormat.Png
+					height = 288
+					width = 512
+				}))
+			}
```

Comment:
Might be worth moving this to an extension function in this class since the same code is used in `getBaseItemAsWatchNextProgram`.

Something like `private fun BaseItemDto.getPosterArtImageUrl(preferParentThumb: Boolean) { ... }`